internal/controllers/gateway/route_predicate_test.go

```diff
@@ -0,0 +1,230 @@
+package gateway
+
+import (
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/samber/lo"
+	"github.com/samber/mo"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/kong/kubernetes-ingress-controller/v3/internal/controllers"
+	"github.com/kong/kubernetes-ingress-controller/v3/internal/gatewayapi"
+	"github.com/kong/kubernetes-ingress-controller/v3/internal/manager/scheme"
+)
+
+func testIsRouteAttachedToReconciledGateway[routeT gatewayapi.RouteT](
+	t *testing.T,
+	cl client.Client,
+	gatewayNN controllers.OptionalNamespacedName,
+	route routeT,
+	expectedResult bool,
+) {
+	logger := logr.Discard()
+
+	result := IsRouteAttachedToReconciledGateway[routeT](cl, logger, gatewayNN, route)
+	require.Equal(t, expectedResult, result)
+}
```

Comment:
I'm just curious: was there a specific reason to add this `func` rather than put it's body inside the `t.Run()`? I guess that's fine to leave it as is but feels odd to put those 3 lines outside of test body.

Reply:
Because the test case struct cannot include a type with type generic, we cannot simply run a 
```
for _, tc := range testCase {
IsRouteAttachedToReconciledGateway(...)
}
```
So I separated the function for running tested function and comparing.
But since we only run tests for `HTTPRoute` here and the comparing code is simple, we can merge them into one function.

Reply:
I missed the fact that `IsRouteAttachedToReconciledGateway` accepts a `client.Object` and does not use the `routeT` as param type. 

👍